fenix/app/src/main/java/org/mozilla/fenix/components/metrics/MetricsMiddleware.kt

```diff
@@ -20,18 +24,39 @@ class MetricsMiddleware(
         next: (AppAction) -> Unit,
         action: AppAction,
     ) {
-        handleAction(action)
+        handleAction(context, action)
         next(action)
     }
 
-    private fun handleAction(action: AppAction) = when (action) {
+    private fun handleAction(context: MiddlewareContext<AppState, AppAction>, action: AppAction) = when (action) {
         is AppAction.AppLifecycleAction.ResumeAction -> {
             metrics.track(Event.GrowthData.SetAsDefault)
             metrics.track(Event.GrowthData.FirstAppOpenForDay)
             metrics.track(Event.GrowthData.FirstWeekSeriesActivity)
             metrics.track(Event.GrowthData.UsageThreshold)
             metrics.track(Event.GrowthData.UserActivated(fromSearch = false))
         }
+
+        is AppAction.AwesomeBarAction.EngagementFinished -> {
+            val interactions = context.state.awesomeBarState.fxSuggestPingsInfo()
+            interactions.forEach {
```

Comment:
Should we be checking for an abandoned engagement before sending the metrics? This check could be somewhere else along this code path if it's needed; no preferences.

```suggestion
        is AppAction.AwesomeBarAction.EngagementFinished -> { action ->
            if (action.abandoned) {
                return // or the equivalent in a middleware.
            }
            val interactions = context.state.awesomeBarState.fxSuggestPingsInfo()
            interactions.forEach {
```